rust/src/data_catalog/unity/models.rs

```diff
@@ -0,0 +1,367 @@
+//! Api models for databricks unity catalog APIs
+
+use std::collections::HashMap;
+
+use serde::Deserialize;
+
+/// Error response from unity API
+#[derive(Deserialize)]
+pub struct ErrorResponse {
+    /// The error code
+    pub error_code: String,
+    /// The error message
+    pub message: String,
+}
+
+/// Get table response
+#[derive(Deserialize)]
+#[serde(untagged)]
+pub enum GetTableResponse {
+    /// Successful response
+    Success(Table),
+    /// Error response
+    Error(ErrorResponse),
+}
```

Comment:
I tested it and unfortunately `Error(ErrorResponse)` is not equivalent to `Error { error_code: String, message: String }`, so right now it fails to parse (the same for `Success(Table)`.

So far I haven't found any other way other than defining the structs inline, like it was before.

I thought the following might work but it didn't.
```
#[derive(Deserialize)]
#[serde(untagged)]
pub enum GetTableResponse {
    /// Successful response
    Success {
        /// Table information
        #[serde(flatten)]
        table: Table,
    },
    /// Error response
    Error {
        /// Error information
        #[serde(flatten)]
        err: ErrorResponse,
    },
}
```

Reply:
hmm .. if we have to we'll do it, but this makes the types a bit annoying to work with.

The example responses from the docs parse fine in the tests though. would it be possible to provide a response body that doe snot parse? Admittedly, the error case I just inferred from your code :).

Reply:
Ok, now I'm even more confused. From the test, it looks like it should work. I don't know what serde is doing but it didn't parse the real response before I changed the definition back.

I'll try to write a test case to reproduce the problem.

Reply:
So, the error cause was different than I thought (see the comment in `struct Table`). At least the definitions don't have to be inlined.